### ISSUE TITLE
ALL_CAPS private const field bypasses the rules

### ISSUE THREAD

**Issue by TheToughGuy** [dotnet/roslyn#52865]
**Version Used**: 
VS 2019 16.9.2
**Steps to Reproduce**:

1. Add style Rule to enforce PascalCase for private const fields
[PascalCaseRule.txt](https://github.com/dotnet/roslyn/files/6366290/PascalCaseRule.txt)

2. In a c# console app add the field `private const string SESSIONDB_CONTEXT_JSON_DATA_PATH = "someData";`

**Expected Behavior**:
should be warned the field is wrongly named
**Actual Behavior**:
VS accepts the field

**Comment by Youssef1313** [comment]
@sharwell This is related to DotNetAnalyzers/StyleCopAnalyzers SA1303. Can you move the issue? Thanks.

**Comment by TheToughGuy** [comment]
Sorry, bit of a nube, how do I move the issue?

**Comment by Youssef1313** [comment]
> Sorry, bit of a nube, how do I move the issue? I think maybe I lack the permissions...

Yes. Only users with write access on both repositories can move issues. You can wait for @sharwell, or close this one and open a new one in [DotNetAnalyzers/StyleCopAnalyzers](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/new).

**Comment by TheToughGuy** [comment]
Moving the issue to DotNetAnalyzers/StyleCopAnalyzers.

**Comment by RehanSaeed** [comment]
This rule tries to replicate a StyleCop rule `SA1303` using `.editorconfig` but no warning is shown. The issue is valid and should be reopened.

```
# All constant fields must be PascalCase
# https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1303.md
dotnet_naming_symbols.stylecop_constant_fields_group.applicable_accessibilities = public, internal, protected_internal, protected, private_protected, private
dotnet_naming_symbols.stylecop_constant_fields_group.required_modifiers         = const
dotnet_naming_symbols.stylecop_constant_fields_group.applicable_kinds           = field
dotnet_naming_rule.stylecop_constant_fields_must_be_pascal_case_rule.symbols    = stylecop_constant_fields_group
dotnet_naming_rule.stylecop_constant_fields_must_be_pascal_case_rule.style      = pascal_case_style
dotnet_naming_rule.stylecop_constant_fields_must_be_pascal_case_rule.severity   = warning
```

**Comment by Youssef1313** [comment]
@RehanSaeed It's opened now in https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3337.
If you meant it should remain in dotnet/roslyn, can you elaborate more? The provided editorconfig is for SA1303, which doesn't live in roslyn.

**Comment by neoKushan** [comment]
@Youssef1313 I think the issue is in the handling of the editorconfig rules. It just so happens that this rule is emulating a rule from Stylecop but it's not a stylecop specific thing, it's the parsing of .editorconfig files and applying those rules that's failing

**Comment by RehanSaeed** [comment]
@neoKushan Yes, that is correct.

**Comment by TheToughGuy** [comment]
reopened as per @RehanSaeed 's comment

**Comment by sharwell** [comment]
Duplicate of #22745
